### PR TITLE
[Merged by Bors] - doc: revamp "Using `mathlib4` as a dependency"

### PR DESCRIPTION
**Pull request by Kha**
---
See comments below

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,29 +52,49 @@ You will need to make a PR after committing the changes to this file.
 
 ## Using `mathlib4` as a dependency
 
-To start a new project that uses mathlib4 as a dependency:
+### In a new project
 
-1. Open a folder that contains no file named `lean-toolchain`.
-2. Update your default toolchain to one that is sufficiently recent with `elan default leanprover/lean4:nightly-2023-02-04`
```
Comment (Kha) [inline README.md:58]: Let's stop using `elan default` and hopefully remove it soon, less global state

File: README.md [hunk 2]
```diff
@@ -52,29 +52,49 @@ You will need to make a PR after committing the changes to this file.
...
+To start a new project that uses mathlib4 as a dependency, run `elan run leanprover/lean4:nightly-2023-02-04 lake new <your_project_name> math`.
+This uses the Lake version with the most recent `new math` implementation independent of your default `elan` toolchain.
+You now have a folder named `your_project_name` that contains a new Lake project.
+The `lakefile.lean` file is configured with the `mathlib4` dependency.
+`lean-toolchain` points to the same version of Lean 4 as used by mathlib4.
+Continue with "Getting started" below.
 
-3. Run `lake new <your_project_name> math`.
-4. You now have a folder named `your_project_name` that contains a new `lake` project. The `lakefile.lean` folder is configured with the `mathlib4` dependency.
-5. Change your current directory to the project folder and run `lake update`. This step downloads `mathlib4` as well as its dependencies.
```
Comment (Kha) [inline README.md:62]: `update` is unnecessary

File: README.md [hunk 3]
```diff
@@ -52,29 +52,49 @@ You will need to make a PR after committing the changes to this file.
...
-6. Run `cp lake-packages/mathlib/lean-toolchain .` to make sure your new project uses the same Lean version as `mathlib4`.
```
Comment (Kha) [inline README.md:63]: already done by `new math`

File: README.md [hunk 4]
```diff
@@ -52,29 +52,49 @@ You will need to make a PR after committing the changes to this file.
...
-7. (Optional) In order to save time compiling all of mathlib and its dependencies, run `lake exe cache get`. This step requires that you have `curl 7.69`  or higher.
-8. Run `lake build`. If you have get no build errors, you are good to go!
```
Comment (Kha) [inline README.md:65]: `lake build` in the default template doesn't even import Mathlib!

File: README.md [hunk 5]
```diff
@@ -52,29 +52,49 @@ You will need to make a PR after committing the changes to this file.
...
+### In an existing project
 
-
-If you already have a project and you want to use `mathlib4`, add these lines to your `lakefile.lean`:
+If you already have a project and you want to use mathlib4, add these lines to your `lakefile.lean`:
 ```
 require mathlib from git
   "https://github.com/leanprover-community/mathlib4" @ "<REVISION>"
 ```
-Where `<REVISION>` can be a commit hash, a branch or a tag. You can check [this section](https://github.com/leanprover/lake/#adding-dependencies) from Lake's README for more info.
```
Comment (Kha) [inline README.md:73]: Basically everyone wants to follow `master` really

File: README.md [hunk 6]
```diff
@@ -52,29 +52,49 @@ You will need to make a PR after committing the changes to this file.
...
+with a similar or larger number.
+Now try adding `import Mathlib` or a more specific import to a project file.
+This should take insignificant time and not rebuild any mathlib files.
+
+### Updating `mathlib4`
 
-Either way, make sure that your project uses the same Lean 4 toolchain as the one used in `mathlib4`.
+Run these commands in sequence:
+```
+lake update
```
Comment (arthurpaulino) [inline README.md:92]: This changes `lake-manifest.json`, causing `cache` not to find the files needed

Comment (Kha) [reply README.md:92]: We won't get around that if the goal is to update mathlib4 :smile:

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**